lib/test_case.dart

```diff
@@ -129,6 +129,23 @@ class TestCase {
     return TestCase.fromJson(response);
   }
 
+  static Future<SoftDeleteTestCase> delete(
+    int caseId, {
+    bool soft = false,
+  }) async {
+    final parametersMap = {
+      'soft': (soft == true) ? 1 : 0,
+    };
+
+    final response = await TestRail.instance.client.request(
+      '/delete_case/$caseId',
+      RequestMethod.post,
+      params: parametersMap,
+    );
+
+    return SoftDeleteTestCase.fromJson(response);
```

Comment:
will `response` always  produce the `SoftDeleteTestCase` structure regardless if `soft` is `true` or `false` ?

Reply:
Perhaps naming should be improved then to match exact functionality?

Reply:
yea, I think we just update the function to return `Future<SoftDeleteTestCase?>` and add `return null` when `soft` is `false`